teapots/choreographer-30fps/src/main/cpp/CMakeLists.txt

```diff
@@ -17,36 +17,53 @@
 cmake_minimum_required(VERSION 3.4.1)
 
 # build native_app_glue as a static lib
-add_library(native_app_glue STATIC
-    ${ANDROID_NDK}/sources/android/native_app_glue/android_native_app_glue.c)
-
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11 -Wall -fno-exceptions -fno-rtti")
+add_library(nativeAppGlue STATIC
+  ${ANDROID_NDK}/sources/android/native_app_glue/android_native_app_glue.c)
 
 # build the ndk-helper library
-set(ndk_helper_dir ../../../../common/ndk_helper)
-add_subdirectory(${ndk_helper_dir} ndk_helper)
-
-# Export ANativeActivity_onCreate(), 
-# Refer to: https://github.com/android-ndk/ndk/issues/381.
-set(CMAKE_SHARED_LINKER_FLAGS
-    "${CMAKE_SHARED_LINKER_FLAGS} -u ANativeActivity_onCreate")
+set(ndkHelperSrc ../../../../common/ndk_helper)
+add_subdirectory(${ndkHelperSrc} ndkHelperBin)
 
 # now build app's shared lib
-add_library(ChoreographerNativeActivity SHARED
+add_library(ChoreographerNativeActivity
+  SHARED
     ChoreographerNativeActivity.cpp
-    TeapotRenderer.cpp)
+    TeapotRenderer.cpp
+)
+
+set_target_properties(ChoreographerNativeActivity
+  PROPERTIES
+    CXX_STANDARD 11
+    CXX_STANDARD_REQUIRED YES
+    CXX_EXTENSIONS NO
+)
+target_compile_features(ChoreographerNativeActivity
+  PRIVATE cxx_noexcept
+)
+target_compile_options(ChoreographerNativeActivity
+  PRIVATE -Wall -Werror -fno-rtti
+)
 
-target_include_directories(ChoreographerNativeActivity PRIVATE
+# Export ANativeActivity_onCreate(),
+# Refer to: https://github.com/android-ndk/ndk/issues/381.
+set(CMAKE_SHARED_LINKER_FLAGS
+  "${CMAKE_SHARED_LINKER_FLAGS} -u ANativeActivity_onCreate")
+
+target_include_directories(ChoreographerNativeActivity
+  PRIVATE
     ${ANDROID_NDK}/sources/android/cpufeatures
     ${ANDROID_NDK}/sources/android/native_app_glue
-    ${ndk_helper_dir})
+    ${ndkHelperSrc}
+)
```

Comment:
Might I suggest here that cpufeatures and native_app_glue include lines are not needed?

The reasoning is CMake will use targets to determine what include directories are required.  i.e. in the target_link_libraries when using native_app_glue and cpufeatures both targets have interface include headers property defined, so CMake is smart enough to allow you to grab the directories for free by simply linking them.
Also, ndkHelperSrc is already included by virtue of the fact that's where the source files are (you only need to provide CMake a custom include directory if it is not the same place as the c++ files are located).
I'd venture this entire target_include_directories is not needed.

Reply:
+1. The spirit of the linked talk is *reducing* unnecessary duplication.